pkg/sync/kubernetes/featureflagconfiguration/clientset.go

```diff
@@ -0,0 +1,128 @@
+	"reflect"
+	"time"
+
+	"github.com/open-feature/flagd/pkg/sync"
+	ffv1alpha1 "github.com/open-feature/open-feature-operator/apis/core/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FFCInterface interface {
+	FeatureFlagConfigurations(namespace string) Interface
+}
+
+type FFCClient struct {
+	restClient rest.Interface
+}
+
+func createFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(obj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("object is not a FeatureFlagConfiguration")
+	}
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeCreate,
+			},
+		}
+	}
+	return nil
+}
+
+func updateFuncHandler(oldObj interface{}, object client.ObjectKey, c chan<- sync.INotify) {
+	if oldObj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeModify,
+			},
+		}
+	}
+}
+
+func deleteFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) {
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeDelete,
+			},
+		}
+	}
+}
+
+func WatchResources(l log.Entry, clientSet FFCInterface, refreshTime time.Duration,
+	object client.ObjectKey, c chan<- sync.INotify,
+) {
+	ns := "*"
+	if object.Namespace != "" {
+		ns = object.Namespace
+	}
+	_, ffConfigController := cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo metav1.ListOptions) (result runtime.Object, err error) {
+				return clientSet.FeatureFlagConfigurations(ns).List(lo)
+			},
+			WatchFunc: func(lo metav1.ListOptions) (watch.Interface, error) {
+				return clientSet.FeatureFlagConfigurations(ns).Watch(lo)
+			},
+		},
+		&ffv1alpha1.FeatureFlagConfiguration{},
+		refreshTime,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if err := createFuncHandler(obj, object, c); err != nil {
+					l.Warn(err.Error())
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				deleteFuncHandler(obj, object, c)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				// This indicates a change to the custom resource
+				// Typically this could be anything from a status field to a spec field
+				// It is important to now assertain if it is an actual flag configuration change
+				updateFuncHandler(oldObj, object, c)
```

Comment:
> It is important to now assertain if it is an actual flag configuration change

I don't see any logic trying to do this (maybe I'm missing something). Is this a place where we'd want to perhaps use something like a hashing mechanism to determine if the actual flag spec changed?

Reply:
I did try to implement it initially but comparing the old/newObject featureflagspec didn't work, there is a time component plus the complexity of the cache refresh that just made it a bit awkward. The current implementation is more naive but works 100%

Reply:
I will remove the comments